packages/contracts-periphery/contracts/universal/op-nft/Optimist.sol

```diff
@@ -0,0 +1,187 @@
+     *          The tokenId is the uint256 of the recipient address.
+     * @param   _recipient  The address of the token recipient.
+     */
+    function mint(address _recipient) public {
+        require(balanceOf(_recipient) == 0, "ALREADY_MINTED");
+        require(isWhitelisted(_recipient), "NOT_WHITELISTED");
+        _safeMint(_recipient, _tokenIdOfOwner(_recipient));
+    }
+
+    /**
+     * @notice Returns the URI for the token metadata.
+     * @dev The token URI will be stored at baseURI + '/' + tokenId + .json
+     * @param tokenId The token ID to query.
+     * @return The URI for the given token ID.
+     */
+    function tokenURI(uint256 tokenId) public view virtual override returns (string memory) {
+        require(ownerOf(tokenId) != address(0), "NOT_MINTED");
+        return
+            string(
+                abi.encodePacked(
+                    baseURI(),
+                    "/",
+                    Strings.toHexString(tokenId, _ADDRESS_LENGTH),
+                    ".json"
+                )
+            );
+    }
+
+    /**
+     * @notice  Returns whether an address is whitelisted
+     * @dev     The whitelist is an attestation by the admin of this contract
+     * @return  boolean  Whether the address is whitelisted
+     */
+    function isWhitelisted(address _recipient) public view returns (bool) {
+        return
+            attestationStation
+                .attestations(owner(), _recipient, bytes32("optimist.can-mint"))
+                .length > 0;
+    }
+
+    /**
+     * @notice  Returns decimal tokenid for a given address
+     * @return  uint256 decimal tokenId
+     */
+    function tokenIdOfOwner(address _owner) public view returns (uint256) {
+        require(balanceOf(_owner) > 0, "NOT_MINTED");
+        return _tokenIdOfOwner(_owner);
+    }
+
+    function _tokenIdOfOwner(address _owner) internal pure returns (uint256) {
+        return uint256(uint160(_owner));
+    }
+
+    /**
+     * @notice  Returns decimal tokenid for a given address
+     * @return  uint256 decimal tokenId
+     */
+    function baseURI() public view returns (string memory) {
+        return
+            string(
+                abi.encodePacked(
+                    attestationStation.attestations(
+                        owner(),
+                        address(this),
+                        bytes32("optimist.base-uri")
+                    )
+                )
+            );
+    }
+
+    /**
+     * @notice  (Internal) Soulbound
+     * @dev     Override internal function to prevent transfers of the Optimist token.
+     * @param   _from  The address of the token sender.
+     * @param   _to    The address of the token recipient.
+     */
+    function _beforeTokenTransfer(
+        address _from,
+        address _to,
+        uint256
+    ) internal virtual override {
+        require(_from == address(0) || _to == address(0), "SBT_TRANSFER");
+    }
+
+    /**
```

Comment:
I don't think this is necessary. Even if someone tries to approve, the other user won't be able to transfer.

Reply:
Preventing approvals does save possible gas with user error

Reply:
Yea this was my thinking
1. Save gas
2. Make user aware transffering isn't possible as early as possible in the process